File: src/fides/api/schemas/masking/masking_strategy_description.py
from typing import List

from pydantic import BaseModel


class MaskingStrategyConfigurationDescription(BaseModel):
    """The description model for a specific configuration in a masking strategy"""

    key: str
    optional: bool = True
    description: str


class MaskingStrategyDescription(BaseModel):
    """The description model for a masking strategy"""

    name: str
    description: str
    configurations: List[MaskingStrategyConfigurationDescription]

File: clients/privacy-center/types/api/models/MaskingStrategyDescription.ts
/* istanbul ignore file */
/* tslint:disable */

import type { MaskingStrategyConfigurationDescription } from "./MaskingStrategyConfigurationDescription";

/**
 * The description model for a masking strategy
 */
export type MaskingStrategyDescription = {
  name: string;
  description: string;
  configurations: Array<MaskingStrategyConfigurationDescription>;
};
